### PR TITLE
LP-1592 Backend implementation for showing badges on Trophycase

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### [LP-1592](https://philanthropyu.atlassian.net/browse/LP-1592)

- Backend implementation for showing badges on trophy case

### CODE REVIEW

File: openedx/features/badging/helpers.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import copy
+
+
+def create_torphycase_data(torphycase_dict, badge_type, all_badges, earned_user_badges, enrolled_courses_data):
```
[the remark this answers is not in the document]
Comment (emraan004) [reply openedx/features/badging/helpers.py]: Good catch

File: openedx/features/badging/helpers.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import copy
+
+
+def create_trophycase_data(trophycase_dict, badge_type, all_badges, earned_user_badges, enrolled_courses_data):
```
Comment (saad-waqar) [inline openedx/features/badging/helpers.py]: To me this method is confusing. Half of the work we're doing in the view even though we're only using the extracted data in this method. If this whole code block is to be used again lets shift the queries in the model into this method. If this isn't to be used again lets shift this to the view.
Moreover lets use a more specific name for this method as to what it does

File: openedx/features/badging/helpers.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    :param enrolled_courses_data: Courses enrolled by user
+    :return: None
+    """
+    for course_id, display_name in enrolled_courses_data:
+
+        # make deepcopy of badges so that earned badge flag can be added for specific course
+        all_badges_copy = copy.deepcopy(all_badges)
+
+        # append data if key exists otherwise
+        # set default value first before appending data
```
Comment (ahmed-faraz46) [inline openedx/features/badging/helpers.py]: Let's, please make this comment more descriptive

File: openedx/features/badging/helpers.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        trophycase_dict[unicode(course_id)] = {
+            'display_name': display_name,
+            'badges': course_badges
+        }
+
+    return trophycase_dict
+
+
+def get_all_badges(badge_queryset, course_id, earned_badges):
```
Comment (saad-waqar) [inline openedx/features/badging/helpers.py]: lets make badge_queryset an optional parameter and run the query in case its not passed

**Pull request merged**